src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -238,3 +252,26 @@ it('should highlight field if required but no value selected', () => {
   expect(required).toBeVisible();
   expect(required).toHaveClass('Mui-error');
 });
+
+it('should show empty list if no options are given', () => {
+  const nullSQFormDropdown = renderDropdown(null, DEFAULT_PROPS);
+
+  renderSQForm(nullSQFormDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', {name: EMPTY_OPTION});
+  userEvent.click(expandButton);
+
+  expect(screen.queryByRole('option')).not.toBeInTheDocument();
+});
+
+it('should show empty value if initial value not in options', () => {
```

Comment:
These tests are looking great!

For this one you can suppress that noisy console warning with a spy:
```js
// mock console.warn
const consoleWarnSpy = jest
  .spyOn(console, 'warn')
  .mockImplementation(() => {});

const initialValues = {
  dropdownExample: 5
};

renderSQForm(renderDropdown(), {initialValues});

const expandButton = screen.queryByRole('button', {name: EMPTY_OPTION});

expect(expandButton).not.toBeInTheDocument();
// check that the expected warning was called
expect(consoleWarnSpy).toHaveBeenCalledWith(
  expect.stringMatching(/requested display value.*could not be found/i)
);
// restore the mock to prevent it leaking into subsequent tests
consoleWarnSpy.mockRestore();
```

This way we can check that the test passes and the appropriate console warning was given while avoiding the clutter in the terminal output

Reply:
Nice! Thanks for this. Added to this test and the other one that also has a console.warn message.